starter-code/src/clue.js

```diff
@@ -85,10 +85,130 @@ name: Patio
+var mrMustard = {first_name:"Jack", last_name:"Mustard", color:"yellow", description:"He is a former football player who tries to get by on his former glory", age:62, image:"https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg", occupation:"retired football player"};
+
 
 // Rooms' Collection
-var roomsArray = [];
+
+var dinningRoom = {name:"Dining Room"};
+var conservatory = {name:"Conservatory"};
+var kitchen= {name:"Kitchen"};
+var study = {name:"Study"};
+var library = {name:"Library"};
+var billiardRoom= {name:"Billiard Room"};
+var lounge = {name:"Lounge"};
+var ballroom = {name:"Ballroom"};
+var hall = {name:"Hall"};
+var spa = {name:"Spa"};
+var livingRoom= {name:"Living Room"};
+var observatory= {name:"Observatory"};
+var theater = {name:"Theater"};
+var guestHouse= {name:"Guest House"};
+var patio= {name:"Patio"};
+
+
 
 // Weapons Collection
+
+
+var rope = {name:"Rope", weight:10};
+var knife = {name:"Knife", weight:8};
+var candlestick = {name:"Candlestick", weight:2};
+var dumbbell = {name:"Dumbbell", weight:30};
+var poison = {name:"Poison", weight:2};
+var axe = {name:"Axe", weight:15};
+var bat = {name:"Bat", weight:13};
+var trophy = {name:"Trophy", weight:25};
+var pistol = {name:"Pistol", weight:20};
+
+
+//////////////////////////////////////// pushing
+
+var charactersArray = [];
+ charactersArray.push(mrGreen);
+ charactersArray.push(drOrchid);
+ charactersArray.push(profPlum);
+ charactersArray.push(missScarlet);
+ charactersArray.push(mrsPeacock);
+ charactersArray.push(mrMustard);
+
+
 var weaponsArray = [];
+  weaponsArray.push(rope);
+  weaponsArray.push(knife);
+  weaponsArray.push(candlestick);
+  weaponsArray.push(dumbbell);
+  weaponsArray.push(poison);
+  weaponsArray.push(axe);
+  weaponsArray.push(bat);
+  weaponsArray.push(pistol);
+
+var roomsArray = [];
+  roomsArray.push(dinningRoom);
+  roomsArray.push(conservatory);
+  roomsArray.push(kitchen);
+  roomsArray.push(study);
+  roomsArray.push(library);
+  roomsArray.push(billiardRoom);
+  roomsArray.push(lounge);
+  roomsArray.push(ballroom);
+  roomsArray.push(hall);
+  roomsArray.push(spa);
+  roomsArray.push(livingRoom);
+  roomsArray.push(observatory);
+  roomsArray.push(theater);
+  roomsArray.push(guestHouse);
+  roomsArray.push(patio);
+
+function randomSelector(array){
+  if (array.length === 0){
+    return undefined;
+  } let randomCard = array[Math.floor(Math.random()* array.length)];
+  // if (array.length === 1){
+    // return array[0];
+    
+  return randomCard;
+}  
+
+
+function pickMistery(){
+  var misteryEnvelope = [];
+  
+  let randomCharacter = randomSelector(charactersArray);
+  let randomWeapon = randomSelector(weaponsArray);
+  let randomRoom = randomSelector(roomsArray);
+
+  misteryEnvelope.push(randomCharacter);
+  misteryEnvelope.push(randomWeapon);
+  misteryEnvelope.push(randomRoom);
+
+  return misteryEnvelope;
+
+
+}
+
+function revealMistery(misteryEnvelope){
+
+  console.log(misteryEnvelope);
```

Comment:
Every variable that you create takes part of the memory so try not to create variables when there's no need. You can just:
```
return `${misteryEnvelope[0].first_name} ${misteryEnvelope[0].last_name} killed Mr.Boddy using the ${misteryEnvelope[1].name} in the ${misteryEnvelope[2].roomNames}!!!!`
```